pkg/auth/providers/saml/saml_provider.go

```diff
@@ -192,14 +256,29 @@ func (s *Provider) saveSamlConfig(config *v3.SamlConfig) error {
 	}
 
 	config.SpKey = common.GetName(config.Type, field)
+	if s.name == ShibbolethName {
+		samlConfig := v3.SamlConfig{}
+		config.DeepCopyInto(&samlConfig)
 
-	logrus.Debugf("updating samlConfig %s", s.name)
-	_, err = s.authConfigs.ObjectClient().Update(config.ObjectMeta.Name, config)
-	if err != nil {
+		ldapConfig, certpool, err := s.getLDAPConfig()
+		if err != nil {
+			return err
+		}
+
+		ldapConfigCopy := v3.LdapConfig{}
+		ldapConfig.DeepCopyInto(&ldapConfigCopy)
+		s.hasValidLdapConfig = ldaputils.ValidateLdapConfig(ldapConfigCopy, certpool)
+		_, err = s.authConfigs.ObjectClient().Update(
+			config.ObjectMeta.Name,
+			&v3.ShibbolethConfig{
+				SamlConfig: samlConfig,
+				LdapFields: ldapConfigCopy.LdapFields,
```

Comment:
maybe we should save ldap fields only if `hasValidLdapConfig` returns true?

Reply:
I can't because they'll get cleared out and dropped from the UI so the user would need to re-enter them all.